lib/galaxy/webapps/galaxy/api/toolshed.py

```diff
@@ -310,3 +310,19 @@ def search(self, trans, **kwd):
             return {}
         response = json.loads(util.url_get(tool_shed_url, params=dict(q=q), pathspec=['api', 'repositories']))
         return response
+
+    @expose_api
+    @web.require_admin
+    def request(self, trans, **params):
+        """
+        GET /api/tool_shed/request
+        """
+        tool_shed_url = params.pop("tool_shed_url")
```

Comment:
I would prefer it if you were keying on something and recovering the URL server side - shouldn't allow the client to make such arbitrary queries on behalf of the server... but I guess it is hidden behind a ``@require_admin`` at least.

Reply:
Actually if you can just verify this value is in the configured tool shed URLs (from tool_sheds_conf.xml - however that gets configured) that would be enough I think to greatly reduce the exploit surface of this.

Reply:
I agree with you, this is a temporary endpoint and will either be replaced by an explicit one or the toolshed service will be setup for cors.

Reply:
I followed up on your recommendation. The endpoint is now validating the toolshed url. Thanks.